Accept el.setAttribute('mixin', '')

takahirox opened the pull request:
**Description:**

This PR enables `el.setAttribute('mixin', '')` (#2280).
It clears all mixins of el.

This PR doesn't touch `removeAttribute('mixin')` because
another guy is working on here #2282

ngokevin commented:
Thanks!

dmarcos commented on tests/core/a-entity.test.js:
```diff
@@ -1045,6 +1045,19 @@ suite('a-entity', function () {
+      var el = this.el;
+      mixinFactory('material', {material: 'shader: flat'});
+      mixinFactory('position', {position: '1 2 3'});
+      el.setAttribute('mixin', 'material position');
```
can we check for `el.mixinEls` too? `assert.equal(el.mixinEls.length, 2)` before and `assert.equal(el.mixinEls.length, 0)` after

takahirox replied:
That's a good idea. I will.

dmarcos replied:
thanks!

takahirox commented:
BTW, I'm thinking to replace `new(old)Mixins.split(' ')` with `new(old)Mixins.trim().split(/\s+/)`.
The latter can accept two or more spaces, tabs, and new lines.
It'd be useful for users.

    (example in the console)
    -> var str = ' abc def\tghi\n jkl  mn  '
    <- undefined
    -> str
    <- " abc def	ghi
         jkl  mn  "
    -> str.split(' ')
    <- ["", "abc", "def	ghi↵", "jkl", "", "mn", "", ""]
    -> str.trim().split(/\s+/)
    <- ["abc", "def", "ghi", "jkl", "mn"]

What do you guys think?

dmarcos commented:
sure!
